### DISCUSSION TITLE
Enforce content types on endpoints

### DISCUSSION THREAD

**Discussion by men6288** [aws-powertools/powertools-lambda-python#3912] [Q&A]
Hey folks,

I'm looking to understand if it's possible to specify a content type for a route using the API gateway rest resolver. If that's not possible, I'm just looking for an easy way to validate the body exists and that it can be converted from json to a dict.

If there is an example, that would be fantastic.

Thanks!

**Comment by leandrodamascena** [comment]
Hey @men6288! Apologies for missing this discussion earlier. Do you still need assistance with this topic?

Regarding the definition of a specific content type for a route using methods like `@app.get/post/patch/others`, it's not directly supported. However, you can utilize the `Response` object to define the content type of that route. 
If you're interested in data validation, you can explore using the `Resolver` with the `enable_validation` flag to validate the type of the request Body. Here's an example: https://docs.powertools.aws.dev/lambda/python/latest/core/event_handler/api_gateway/#validating-payload-subset

Please let me know any other question.

Thanks

**Reply by men6288** [reply]
Thanks @leandrodamascena!

Has there been any consideration of enforcing this?

**Reply by leandrodamascena** [reply]
Hello @men6288! I didn't understand the question well. Could you provide some more details or clarify a little more?

Thanks

**Reply by men6288** [reply]
For example, in chalice they have the ability to do something like this under request content types:
```
@app.route('/', methods=['POST'],
           content_types=['application/x-www-form-urlencoded'])
```


https://aws.github.io/chalice/tutorials/basicrestapi.html

Which will return a bad error if the content type is not the same. In theory, couldn't a middleware be created to validate this?
